README.md

```diff
@@ -6,5 +6,73 @@
 Virtual Machine agent for hardware virtualized containers
 
 ## Role
-This project holds the code related to a virtual machine agent relying on the communication protocol defined by hyperstart. That way, it allows to spawn some processes on behalf of pod/container(s) running inside the virtual machine.
-The code relies heavily on [libcontainer](https://github.com/opencontainers/runc/tree/master/libcontainer) so that we can reuse as much as possible the code used by `runc` (standard to run containers on bare metal).
+[`cc-agent`](https://github.com/clearcontainers/agent) is a daemon running in the
+guest as a supervisor for managing containers and processes running within
+those containers.
+
+The `cc-agent` execution unit is the pod. A `cc-agent` pod is a container sandbox
+defined by a set of namespaces (NS, UTS, IPC and PID). `cc-runtime` can run several
+containers per pod to support container engines that require multiple containers
+running inside a single VM. In the case of docker, `cc-runtime` creates a single
+container per pod.
+
+`cc-agent` uses a communication protocol defined by the [hyperstart](https://github.com/hyperhq/hyperstart)
+project. This was chosen to maintain backward compatibility with the `hyperstart`
+agent used in the 2.x `Clear Containers` architecture.
+
+The `cc-agent` interface consists of:
+- A control serial channel over which the `cc-agent` sends and receives specific
+  commands for controlling and managing pods and containers. Detailed information
+  about the commands can be found at [`cc-agent` API](https://github.com/clearcontainers/agent/tree/master/api).
+- An I/O serial channel for passing the container processes output streams (`stdout`,
+  `stderr`) back to `cc-proxy` and receiving the input stream (`stdin`) for them. As
+  all streams for all containers are going through one single serial channel, the
+  `cc-agent` prepends them with container specific sequence numbers. There are
+  at most two sequence numbers per container process: one for `stdout` and `stdin`,
+  and another one for `stderr`.
+
+`cc-agent` supports the following commands:
+- `StartPodCmd`: Sets up a pod in a newly created VM. 
+- `NewContainerCmd`: Creates a new container within a pod. This needs to be sent
+  after the `StartPodCmd` has been issued for starting a pod. This command also
+  starts the container process.
+- `ExecCmd`: Executes a new process within an already running container.
+- `KillContainerCmd`: `cc-shim` uses this to send signals to a container process.
```

Comment:
This definition doesn't indicate that you kill anything. Assuming the command does kill something, I would suggest a rewrite: "Kills a container process. `cc-shim` uses this command to send signals to the container process."